src/initialization.js

```diff
@@ -0,0 +1,286 @@
+const Apify = require('apify');
+const _ = require('lodash');
+const HeaderGenerator = require('header-generator');
+const { TYPES, LABELS } = require('./constants');
+
+const fns = require('./functions');
+
+const { utils: { log } } = Apify;
+
+const {
+    getUrlData,
+    extendFunction,
+    isOverItems,
+    minMaxDates,
+    // eslint-disable-next-line no-unused-vars
+    createGetSimpleResult,
+} = fns;
+
+/**
+ * Throws error if the provided input is invalid.
+ * @param {{ search: String, startUrls: any[], zpids: any[] }} input
+ */
+const validateInput = (input) => {
+    if (!(input.search && input.search.trim().length > 0) && !input.startUrls && !input.zpids) {
+        throw new Error('Either "search", "startUrls" or "zpids" attribute has to be set!');
+    }
+};
+
+/**
+ *
+ * @param {{ minDate: string, maxDate: string }} input
+ * @returns minMaxDate
+ */
+const initializeMinMaxDate = ({ minDate, maxDate }) => {
+    return minMaxDates({
+        min: minDate,
+        max: maxDate,
+    });
+};
+
+/**
+ *
+ * @param {{ search: string, startUrls: any[], type: string, zpids: any[] }} input
+ * @returns startUrls
+ */
+const getInitializedStartUrls = async (input) => {
+    /**
+     * @type {Apify.RequestOptions[]}
+     */
+    const startUrls = [];
+
+    if (input.search && input.search.trim()) {
+        const term = input.search.trim();
+
+        startUrls.push({
+            url: 'https://www.zillow.com',
+            uniqueKey: `${term}`,
+            userData: {
+                label: LABELS.SEARCH,
+                term,
+            },
+        });
+    }
+
+    if (input.startUrls && input.startUrls.length) {
+        if (input.type) {
+            log.warning(`Input type "${input.type}" will be ignored as the value is derived from start url.
+             Check if your start urls match the desired home status.`);
+        }
+
+        const requestList = await Apify.openRequestList('STARTURLS', input.startUrls);
+
+        let req;
+        while (req = await requestList.fetchNextRequest()) { // eslint-disable-line no-cond-assign
+            if (!req.url.includes('zillow.com')) {
+                throw new Error(`Invalid startUrl ${req.url}`);
+            }
+
+            startUrls.push({
+                url: req.url,
+                userData: getUrlData(req.url),
+            });
+        }
+    }
+
+    if (input.zpids && input.zpids.length) {
+        startUrls.push({
+            url: 'https://www.zillow.com/',
+            uniqueKey: 'ZPIDS',
+            userData: {
+                label: LABELS.ZPIDS,
+            },
+        });
+    }
+
+    return startUrls;
+};
+
+const getInitilizedHeaderGenerator = () => {
+    return new HeaderGenerator({
+        browsers: [
+            { name: 'chrome', minVersion: 87 },
+        ],
+        devices: [
+            'desktop',
+        ],
+        operatingSystems: process.platform === 'win32'
```

Comment:
This is excellent thinking! I am not aware of the request anatomy differences from different oses except for the tls ciphers. Does it make any difference for this particular website?

Reply:
I just moved this one from `main` into `initialization`, don't know what was the thinking behind specifying OS here 🤔